srtcore/api.cpp

```diff
@@ -1579,6 +1611,21 @@ int CUDTUnited::groupConnect(CUDTGroup* pg, SRT_SOCKGROUPCONFIG* targets, int ar
             // If m_IncludedGroup is not NULL, the m_IncludedIter is still valid.
             CUDTGroup::gli_t f = ns->m_IncludedIter;
 
+            // Now under a group lock, we need to make sure the group isn't being closed
+            // in order not to add a socket to a dead group.
+            if (g.m_bClosing)
+            {
+                LOGC(aclog.Error, log << "groupConnect: group deleted while connecting; breaking the process");
+
+                // Set the status as pending so that the socket is taken care of later.
+                // Note that all earlier sockets that were processed in this loop were either
+                // set BROKEN or PENDING.
+                f->sndstate = SRT_GST_PENDING;
+                f->rcvstate = SRT_GST_PENDING;
+                retval = -1;
```

Comment:
The socket already references the group, doesn't it?
`ns->m_IncludedIter = f;` 127 lines above.

Reply:
Yes, but there were mutexes unlocked in the meantime and this could have changed. Prolonged use of `f` here has caused a crash in one test.